codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/OperationRegistryGenerator.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.rust.codegen.rustlang.*
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingResolver
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpTraitHttpBindingResolver
+import software.amazon.smithy.rust.codegen.smithy.protocols.ProtocolContentTypes
+import software.amazon.smithy.rust.codegen.util.inputShape
+import software.amazon.smithy.rust.codegen.util.outputShape
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+
+/**
+ * OperationRegistryGenerator
+ */
+class OperationRegistryGenerator(
+    codegenContext: CodegenContext,
+    private val operations: List<OperationShape>,
+) {
+    private val serverCrate = "aws_smithy_http_server"
+    private val service = codegenContext.serviceShape
+    private val operationNames = operations
+        .map { operation -> service.getContextualName(operation).toSnakeCase() }
+    private val model = codegenContext.model
+    private val symbolProvider = codegenContext.symbolProvider
+    private val runtimeConfig = codegenContext.runtimeConfig
+
+    // TODO Is using this like this fine?
+    private val httpBindingResolver: HttpBindingResolver =
+        HttpTraitHttpBindingResolver(codegenContext.model, ProtocolContentTypes.consistent("application/json"))
+
+    fun render(writer: RustWriter) {
+        fromRequestIntoResponseWorkaround(writer)
+
+        Attribute.Derives(setOf(RuntimeType.Debug, RuntimeType.DeriveBuilder)).render(writer)
+        // TODO: Should I use `raw` here? I would not have to escape the `#`.
+        writer.rust("##[builder(pattern = \"owned\")]")
```

Comment:
you can actually just `Attribute.Custom(...).render()`. Raw is also fine, totally up to you